### PR TITLE
Add pre-commit config

### PR DESCRIPTION
**Pull request by oddstr13**
Run `dotnet format`

### CODE REVIEW

**Review by Bond-009 — CHANGES_REQUESTED** [review]
Is this removing or adding the BOM? And I don't like the indentation change after the lock statement.

**Comment by oddstr13** [comment]
> Is this removing or adding the BOM? And I don't like the indentation change after the lock statement.

This is removing BOM, and I agree on the lock statement, if you know how to make `dotnet format` not do that that'd be great

**Comment by oddstr13** [comment]
I couldn't figure out why it didn't add braces there, so I ended up adding them myself.
The indentation it did is technically correct, as it did the equivalent of this;
```csharp
if (true)
foo();
```
to
```csharp
if (true)
    foo();
```
But it should also have wrapped it in `{}` there.

**Review by Bond-009 — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    -   id: check-case-conflict
+    -   id: check-merge-conflict
+    -   id: check-symlinks
+    -   id: destroyed-symlinks
+    -   id: check-vcs-permalinks
+    -   id: detect-private-key
+    -   id: no-commit-to-branch
+
+-   repo: https://github.com/jvstein/pre-commit-dotnet-format
+    rev: 'v0.2'
```
Comment (h1dden-da3m0n) [inline .pre-commit-config.yaml:21]: well this PR was open for long enough for this [project to get deprecated](https://github.com/jvstein/pre-commit-dotnet-format#deprecated)

Comment (oddstr13) [reply .pre-commit-config.yaml:21]: Looks like the deprecating project does not yet have a release, and the head branch depends on .NET 6 preview.

**Review by h1dden-da3m0n — APPROVED** [review]

**Pull request merged**